### ISSUE TITLE
About adding `equals()` method to the `Color` interface

### ISSUE THREAD

**Issue by KennedyTedesco** [php-tui/php-tui#184]
In the current implementation of `PhpTermBackend` and `Style`, we use weak comparisons like:

```php
if ($update->cell->fg != $fg) {
```

However, in the next version of PHPStan (version 2, currently bleedingEdge), weak comparisons will not be allowed unless explicitly ignored.

So, I was thinking about adding this `equals()` to the `Color` interface.

The only drawback I've noticed is that introducing these methods results in approximately a 6% regression in bench tests (due to a new method call, etc.).

**Comment by dantleech** [comment]
Probably not a bad thing considering it can either be an enum or a VO - but it's also a hot point (the buffer diff) so 🤷

**Comment by KennedyTedesco** [comment]
Yeah, 6% regression is something to consider. Also, as you said, it hits a really hot spot. This is definitely not important right now, we can just hold on and see at a later point if it will make sense to change this. :thinking:
